packages/runtime/src/widgets/DateRange/DateRange.tsx

```diff
@@ -0,0 +1,296 @@
+} & EnsembleWidgetProps &
+  FormInputProps<string>;
+
+export const DateRange: React.FC<DateProps> = (props) => {
+  const [startDate, setStartDate] = useState<string>("");
+  const [endDate, setEndDate] = useState<string>("");
+  const [value, setValue] = useState<dayjs.Dayjs | string | undefined>("");
+  const [openPicker, setOpenPicker] = useState(false);
+  const [isCalendarOpen, setIsCalendarOpen] = useState(false);
+  const [enteredDate, setEnteredDate] = useState("Start Date - End Date");
+  const [errorText, setErrorText] = useState("");
+  const [endDateErrorText, setEndDateErrorText] = useState("");
+  const [calendarMonth, setCalendarMonth] = useState<number>(
+    value ? dayjs(value).month() + 1 : dayjs().month() + 1,
+  );
+  const [calendarYear, setCalendarYear] = useState<number>(
+    value ? dayjs(value).year() : dayjs().year(),
+  );
+  const datePickerRef = useRef<HTMLDivElement | null>(null);
+  const action = useEnsembleAction(props.onChange);
+
+  const { values } = useRegisterBindings({ ...props, value }, props.id, {
+    setValue,
+  });
+
+  const onChangeCallback = useCallback(
+    (date: string) => {
+      if (!action) {
+        return;
+      }
+      action.callback({
+        [props?.id as string]: {
+          value: date,
+          setValue,
+          ...props,
+        },
+      });
+    },
+    [action, props],
+  );
+
+  const onDateChange = (date: unknown): void => {
+    if (!isCalendarOpen || !openPicker) {
+      return;
+    }
+
+    const formattedDate = dayjs(date as string).format(DateDisplayFormat);
+    if (formattedDate !== "Invalid Date") {
+      if (
+        isEmpty(startDate) ||
+        (!isEmpty(startDate) && !isEmpty(endDate)) ||
+        dayjs(date as string).isBefore(dayjs(startDate))
+      ) {
+        setStartDate(formattedDate);
+        setEndDate("");
+        setEnteredDate(`${formattedDate} - End Date`);
+      } else {
+        setEndDate(formattedDate);
+        setEnteredDate(`${startDate} - ${formattedDate}`);
+        onChangeCallback(formattedDate);
+      }
+    }
+  };
+
+  const onDateClear = (): void => {
+    setValue(undefined);
+    setStartDate("");
+    setEndDate("");
+    setEnteredDate("Start Date - End Date");
+  };
+
+  const onMonthOrYearChange = (date: unknown): void => {
+    setCalendarMonth(dayjs(date as string | dayjs.Dayjs).month() + 1);
+    setCalendarYear(dayjs(date as string | dayjs.Dayjs).year());
+  };
+
+  useEffect(() => {
+    if (!isDateValid(startDate) && !isDateValid(endDate)) {
+      return;
+    }
+
+    const startDay = parseInt(startDate.substring(3, 5));
+    const endDay = parseInt(endDate.substring(3, 5));
+    const startMonth = parseInt(startDate.substring(0, 2));
+    const endMonth = parseInt(endDate.substring(0, 2));
+    const startYear = parseInt(startDate.substring(6, 10));
+    const endYear = parseInt(endDate.substring(6, 10));
+
+    datePickerRef?.current
+      ?.querySelectorAll(".MuiPickersDay-root")
```

Comment:
Still using `querySelectorAll` to add my custom class in order change the styles when a date is selected

<img width="1141" alt="Screenshot 2023-12-19 at 9 06 04 PM" src="https://github.com/EnsembleUI/ensemble-react/assets/32386713/872b0a2a-fe4d-4f25-91b4-6ebc406a49a8">